plugins/formal-verification/formver.plugin/src/org/jetbrains/kotlin/formver/reporting/ErrorReporter.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.formver.reporting
+
+import org.jetbrains.kotlin.KtSourceElement
+import org.jetbrains.kotlin.contracts.description.EventOccurrencesRange
+import org.jetbrains.kotlin.diagnostics.DiagnosticReporter
+import org.jetbrains.kotlin.diagnostics.reportOn
+import org.jetbrains.kotlin.fir.analysis.checkers.context.CheckerContext
+import org.jetbrains.kotlin.formver.PluginErrors
+import org.jetbrains.kotlin.formver.embeddings.SourceRole
+import org.jetbrains.kotlin.formver.embeddings.SourceRole.ParamFunctionLeakageCheck.fetchLeakingFunction
+import org.jetbrains.kotlin.formver.viper.errors.VerificationError
+
+sealed interface ErrorReporter {
```

Comment:
Nit: I would expect an error reporter to take an error, not a different reporter.  This sounds more like a `FormattedError` or `ReportableError` class or something.

Reply:
I see what you mean! I think `FormattedError` sounds good, I will rename it to that.